### PR TITLE
Added extrinsic set_punishment_delay to vault-registry pallet

### PR DESCRIPTION
**Pull request by RustNinja**
PR related to #305
- introduce new extrinsic `set_punishment_delay` to update `PunishmentDelay` storage value. 
- added unit tests
- added benchmarking
- generated new default_weights for new extrinsic.

### CODE REVIEW

File: pallets/vault-registry/src/tests.rs [hunk 1]
```diff
@@ -129,6 +129,26 @@ fn create_sample_vault_and_issue_tokens(issue_tokens: u128) -> DefaultVaultId<Te
 	create_vault_and_issue_tokens(issue_tokens, DEFAULT_COLLATERAL, DEFAULT_ID)
 }
 
+#[test]
+fn should_set_punishment_delay() {
```
Comment (ebma) [inline pallets/vault-registry/src/tests.rs:133]: Nit: I'd rename to `set_punishment_delay_works()`

File: pallets/vault-registry/src/tests.rs [hunk 2]
```diff
@@ -129,6 +129,26 @@ fn create_sample_vault_and_issue_tokens(issue_tokens: u128) -> DefaultVaultId<Te
...
+	run_test(|| {
+		let punishment_delay = 99;
+		assert_ok!(VaultRegistry::set_punishment_delay(RuntimeOrigin::root(), punishment_delay));
+		assert_eq!(PunishmentDelay::<Test>::get(), punishment_delay);
+	})
+}
+
+#[test]
+fn should_failed_set_punishment_delay() {
```
Comment (ebma) [inline pallets/vault-registry/src/tests.rs:142]: Nit: I'd rename to `set_punishment_delay_fails_for_wrong_origin()`

File: pallets/vault-registry/src/benchmarking.rs [hunk 3]
```diff
@@ -151,6 +151,14 @@ benchmarks! {
 		Oracle::<T>::_set_exchange_rate(vault_id.clone().account_id, get_wrapped_currency_id(), UnsignedFixedPoint::<T>::checked_from_rational(10, 1).unwrap()).unwrap();
 		VaultRegistry::<T>::liquidate_vault(&vault_id).unwrap();
 	}: _(RawOrigin::Signed(vault_id.account_id), vault_id.currencies.clone())
+
+	set_punishment_delay {
+		let punishment_delay : T::BlockNumber = T::BlockNumber::default();
```
Comment (ebma) [inline pallets/vault-registry/src/benchmarking.rs:156]: Nit: There should not be a space character after `let punishment_delay`

File: pallets/vault-registry/src/benchmarking.rs [hunk 4]
```diff
@@ -151,6 +151,14 @@ benchmarks! {
...
+	}: _(RawOrigin::Root, punishment_delay)
+	verify {
+		let punishment_delay : T::BlockNumber = T::BlockNumber::default();
```
Comment (ebma) [inline pallets/vault-registry/src/benchmarking.rs:159]: Same here

**Review by ebma — CHANGES_REQUESTED** [review]
Looks good overall, I just requested some nitpicky things. Apparently CI fails because you have to re-generate the metadata, so I would suggest you run the testchain locally and use subxt to regenerate the metadata-standalone.scale file. After that the CI job should work.

File: pallets/vault-registry/src/tests.rs (hunk 1 above)
Comment (RustNinja) [reply pallets/vault-registry/src/tests.rs:133]: Done.

File: pallets/vault-registry/src/benchmarking.rs (hunk 3 above)
Comment (RustNinja) [reply pallets/vault-registry/src/benchmarking.rs:156]: Done

**Pull request merged**